.github/workflows/ci.yml

```
@@ -0,0 +1,82 @@
+      run: npm ci
+
+    - name: Build project
+      run: npm run build
+      env:
+        # Provide dummy values for build-time environment variables
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN || 'dummy-token' }}
+        GITHUB_REPO_OWNER: ${{ github.repository_owner }}
+        GITHUB_REPO_NAME: ${{ github.event.repository.name }}
+
+    - name: Upload build artifacts
+      uses: actions/upload-artifact@v3
```

actionlint (1.7.7)

the runner of "actions/upload-artifact@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)